R/animint.R

```diff
@@ -1684,6 +1692,28 @@ getLegendList <- function(plistextra){
     geom.unique.rows <- sapply(geom.unique.list, nrow)
     is.ignored <- 1 < geom.data.rows & geom.unique.rows == 1
     gdefs[[leg]]$geoms <- geom.list[!is.ignored]
+    
+    ## Add geom names to be used in getLegend function
+    if(length(geom.name.list) > 0){
+      len.list <- length(geom.list)
+      geom.name.list <- geom.name.list[1:len.list]
+    }
+    # only use geoms not ignored earlier
+    geom.name.list <- geom.name.list[!is.ignored]
+    
+    ## guides_geom() earlier returned geom names that are not
+    ## reflected by the ggtype function
+    ## convert these to get proper geom names for legends
```

Comment:
`guides_geom()` now returns a `draw_key()` method instead of a geom name. So I had to get the geom names from the layers. But the geom name returned by the `guides_geom()` is not the same as the one in the layers, hence I had to resort to this for loop. Will have to keep updating for more cases unless we find a better solution to this.